extensions/api/blocking_api.py

```diff
@@ -39,12 +66,23 @@ def do_GET(self):
             })
 
             self.wfile.write(response.encode('utf-8'))
+            
+        elif self.path.startswith('/api/v1/characters'):
+            self.simple_json_results(get_available_characters()[1:])
         else:
             self.send_error(404)
 
     def do_POST(self):
         content_length = int(self.headers['Content-Length'])
-        body = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        try:
+            body = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        except Exception as e:
+            print(e)
+            self.send_error(400)
+            return
+
+        if shared.args.auth_api:
```

Comment:
Should this be at the top of the `do_POST` function?

Reply:
Yyyy

Reply:
Ok